hbase-server/src/test/java/org/apache/hadoop/hbase/client/SimpleScanResultConsumerImpl.java

```diff
@@ -71,6 +72,29 @@ public synchronized List<Result> getAll() throws Exception {
 
   @Override
   public ScanMetrics getScanMetrics() {
-    return scanMetrics;
+    if (scanMetricsByRegion != null) {
```

Comment:
We've seen this logical structure repeated over and over. Is it worth putting into a utility method on a base-class?